docs/topics/security.txt

```diff
@@ -81,6 +107,202 @@ HSTS for supported browsers.
 Be very careful with marking views with the ``csrf_exempt`` decorator unless
 it is absolutely necessary.
 
+.. _how-csrf-works:
+
+How it works
+------------
+
+Django's CSRF protection places a hidden field on its forms, that will be
+submitted when the form is POSTed. At the same time, it sets a CSRF cookie with
+a matching value.
+
+When Django receives the form, along with the cookie, if the cookie and the
+hidden field don't match, the POST request will be rejected.
+
+This helps distinguish between legitimate (in which the user really did intend
+to submit such a form) and forged (in which a script or other mechanism is used
+to submit a malicious form using the victim's identity) requests.
+
+The technical details of this protection:
+
+* Django creates a random secret value, which only the server will have access
+  to.
```

Comment:
Technically not true since the user has access to it too

Reply:
I understood that the secret really is secret to the server, and everything else (cookies, tokens, all values that are actually passed around in the HTTP cycle) is *generated from* the secret. Is this not the case?